app/src/main/java/com/nervousfish/nervousfish/activities/WaitActivity.java

```diff
@@ -55,24 +52,33 @@ protected void onCreate(final Bundle savedInstanceState) {
         final TextView waitingMessage = (TextView) findViewById(R.id.waiting_message);
         waitingMessage.setText(message);
 
-        LOGGER.info("WaitActivity created");
+        LOGGER.info("Activity created");
     }
 
-    private void evaluateData() {
-        LOGGER.info("Evaluating data");
-        final Intent intent = new Intent(this, MainActivity.class);
-        intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, serviceLocator);
-        intent.putExtra(ConstantKeywords.SUCCESSFUL_BLUETOOTH, true);
-        this.startActivity(intent);
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void onStart() {
+        super.onStart();
+        this.serviceLocator.registerToEventBus(this);
+
+        if (this.dataReceived != null && this.tapCombination != null) {
+            evaluateData();
```

Comment:
I might add it if anyone requests any changes 😉 Otherwise I'm not going to create a commit just for that